### PR TITLE
vse-bude-task#10: added email service

### PR DESCRIPTION
**Pull request by vladyslav-soltanovskyi**
I made an email service and added the **sendGreetingTo** method for an example of sending messages to the mailI made an email service and added the sendGreetingTo method for an example of sending messages to the mail

### CODE REVIEW

File: packages/backend/.env.example [hunk 1]
```diff
@@ -1,2 +0,0 @@
-DATABASE_URL="postgresql://postgres:postgres@localhost:5432/vsebude_db?schema=public"
```
Comment (Vitalii4as) [inline packages/backend/.env.example:1]: Why you deleted .env.example? 
It should exist. Also add here envs you used

File: packages/backend/src/repositories/mailer.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { TransactionalEmailsApi, TransactionalEmailsApiApiKeys } from 'sib-api-v3-typescript';
+import { SendSmtpEmail as ISendSmtEmail } from 'sib-api-v3-typescript'
+import { getEnv } from '@helpers';
+
+export class MailerRepository {
```
Comment (Vitalii4as) [inline packages/backend/src/repositories/mailer.ts:5]: Create interface `IEmailProvider`
Create file `src/providers/email/SendInBlueEmailProvider.ts` and move all this to that file. implement IEmailProvider.

File: packages/backend/src/repositories/index.ts [hunk 3]
```diff
@@ -1,12 +1,14 @@
 import type { PrismaClient } from '@prisma/client';
 import { RandomDataRepository } from './random-data';
 import { UserRepository } from './user';
+import { MailerRepository } from './mailer';
 
 export const initRepositories = (prismaClient: PrismaClient) => ({
   randomDataRepository: new RandomDataRepository(),
   userRepository: new UserRepository(prismaClient),
+  mailerRepository: new MailerRepository(),
```
Comment (Vitalii4as) [inline packages/backend/src/repositories/index.ts:9]: Remove It

File: packages/backend/src/services/index.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
 import type { Repositories } from '@repositories';
 import { RandomDataService } from './random-data';
 import { UserService } from './user';
+import { MailerService } from './mailer';
 
 export const initServices = (repositories: Repositories) => ({
   randomDataService: new RandomDataService(repositories.randomDataRepository),
   userService: new UserService(repositories.userRepository),
+  mailerService: new MailerService(repositories.mailerRepository),
```
Comment (Vitalii4as) [inline packages/backend/src/services/index.ts:9]: just emailService

File: packages/backend/src/services/mailer.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import type { MailerRepository } from '@repositories';
+import { getEnv } from '@helpers';
+
+export class MailerService {
```
Comment (Vitalii4as) [inline packages/backend/src/services/mailer.ts:4]: Just EmailService

File: packages/backend/src/services/mailer.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    private _mailerRepository: MailerRepository;
+
+    constructor(mailerRepository: MailerRepository) {
```
Comment (Vitalii4as) [inline packages/backend/src/services/mailer.ts:7]: Recieve IEmailProvider

File: packages/backend/src/services/mailer.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        this._mailerRepository = mailerRepository;
+    }
+
+    public async sendGreetingTo(email: string) {
```
Comment (Vitalii4as) [inline packages/backend/src/services/mailer.ts:11]: Just `send` method with email and body (or content) parameters

File: packages/backend/src/services/index.ts [hunk 8]
```diff
@@ -1,12 +1,14 @@
 import type { Repositories } from '@repositories';
 import { RandomDataService } from './random-data';
 import { UserService } from './user';
+import { MailerService } from './mailer';
 
 export const initServices = (repositories: Repositories) => ({
```
Comment (Vitalii4as) [inline packages/backend/src/services/index.ts:6]: Create provider instance here and pass it to the EmailService

File: packages/backend/src/repositories/mailer.ts (hunk 2 above)
Comment (Vitalii4as) [reply packages/backend/src/repositories/mailer.ts:5]: We need interface to easily change provider (for example from SendInBlue to SendGrid or any other)

**Review by Vitalii4as — CHANGES_REQUESTED** [review]

File: packages/backend/src/services/index.ts (hunk 4 above)
Comment (Nightlngale) [reply packages/backend/src/services/index.ts:9]: emailSenderService

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/.env.example [hunk 9]
```diff
@@ -1,13 +1,15 @@
 DATABASE_URL="postgresql://postgres:postgres@localhost:5432/vsebude_db?schema=public"
 PORT=3001
 
+# Email envs
+EMAIL_SERVICE_API_KEY=
+EMAIL_SERVICE_ADDRESS=
```
Comment (Vitalii4as) [inline packages/backend/.env.example:6]: Why you added `EMAIL_SERVICE_ADDRESS`? You don't use it. Is it required for SDK?

File: packages/backend/src/common/types/providers/IEmailProvider.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import type { SendSmtpEmail, CreateSmtpEmail } from 'sib-api-v3-typescript';
+import type { IncomingMessage } from 'http';
+
+export interface IEmailProvider {
```
Comment (Vitalii4as) [inline packages/backend/src/common/types/providers/IEmailProvider.ts:4]: Please use for interface custom classes. It needed to easily change the provider. Return just Promise<boolean> - either sent successfully or not. As arguments it should require only email of a user and actual message

**Review by Vitalii4as — CHANGES_REQUESTED** [review]

File: packages/backend/src/providers/email/SendInBlueEmailProvider.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+  constructor() {
+    this._apiInstance.setApiKey(
+      TransactionalEmailsApiApiKeys.apiKey,
+      getEnv('EMAIL_SERVICE_API_KEY'),
+    );
+  }
+
+  public async send(options: EmailOptions) {
+    return await this._apiInstance.sendTransacEmail({
```
Comment (Vitalii4as) [inline packages/backend/src/providers/email/SendInBlueEmailProvider.ts:19]: return `Promise<boolean>` - either sent successfully or not.

**Review by Vitalii4as — APPROVED** [review]

**Pull request merged**